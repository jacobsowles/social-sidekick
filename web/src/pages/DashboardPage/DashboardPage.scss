.dashboard {
  .Collapsible {
    margin-bottom: 10px;

    // TODO: these styles should be tied to the icon, not the collapsible.
    &.facebook-collapsible svg {
      color: #3b5998;
    }

    &.github-collapsible svg {
      color: #333333;
    }

    &.twitter-collapsible svg {
      color: #55acee;
    }

    .Collapsible__trigger svg {
      width: 42px;
    }
  }
}
