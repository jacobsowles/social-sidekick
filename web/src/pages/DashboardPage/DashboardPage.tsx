import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React, { FunctionComponent } from 'react';
import { Link } from 'react-router-dom';

import Collapsible from '@components/Collapsible';
import ContentBox from '@components/ContentBox';
import LoadingSpinner from '@components/LoadingSpinner';
import PageHeader from '@components/PageHeader';
import { UserService } from '@core/types';
import './DashboardPage.scss';

export interface DashboardPageProps {
  serviceModules: any[] | null;
  services: UserService[] | null;
  userId?: string;
}

const DashboardPage: FunctionComponent<DashboardPageProps> = ({ serviceModules }) => {
  return <div className="dashboard">{getDashboardContent(serviceModules)}</div>;
};

const getDashboardContent = (serviceModules: any[]) => {
  if (!serviceModules) {
    return <LoadingSpinner />;
  }

  if (serviceModules.length === 0) {
    return (
      <ContentBox>
        <PageHeader title="Getting Started" />
        <p>
          Once you <Link to="/connections">connect your social media accounts</Link>, you'll be able
          to manage your profile data on this page.
        </p>
      </ContentBox>
    );
  }

  return serviceModules.map((serviceModule: any, key: any) => (
    <Collapsible.Box className={`${serviceModule.name.toLowerCase()}-collapsible`} key={key}>
      <Collapsible.Trigger>
        <FontAwesomeIcon icon={['fab', serviceModule.name.toLowerCase()]} size="3x" />
        <h1>{serviceModule.name}</h1>
      </Collapsible.Trigger>
      <Collapsible.Body>{serviceModule.component}</Collapsible.Body>
    </Collapsible.Box>
  ));
};

export default DashboardPage;
