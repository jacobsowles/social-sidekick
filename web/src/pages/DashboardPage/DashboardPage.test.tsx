import React from 'react';

import { init, shallow, shallowTopLevelElement, toJson } from '@tests/component-test-base';
import DashboardPage, { DashboardPageProps } from '@pages/DashboardPage';

describe('DashboardPage', () => {
  let props: DashboardPageProps;
  init(() => <DashboardPage {...props} />, 'div');

  beforeEach(() => {
    props = { serviceModules: null, services: null, userId: undefined };
  });

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });

  it('should only include the default class name', () => {
    shallowTopLevelElement()
      .prop('className')
      .should.equal('dashboard');
  });
});
