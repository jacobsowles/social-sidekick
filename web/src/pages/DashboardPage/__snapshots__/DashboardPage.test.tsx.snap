// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardPage should render the component without crashing 1`] = `
<div
  className="dashboard"
>
  <LoadingSpinner
    size="4x"
  />
</div>
`;
